### ISSUE TITLE
Playback Modifiers (Play selected parts and/or range)

### ISSUE THREAD

**Issue by Arnesfield** [musescore/MuseScore#24800]
### Your idea

Currently, MuseScore 4 either plays (1) all unmuted/solo parts or (2) only parts that have at least one of their measures selected. With this feature request however, `space` would always play all unmuted/solo parts regardless of selection, and to play only selected parts will require an additional modifier (i.e. `ctrl` + `space`).

> Also, this is mostly the same idea as https://github.com/musescore/MuseScore/issues/23831 but with additional behavior.

First, let's assign some **playback modifiers** (added to `space` key):
- `ctrl` - Play only selected parts
- `shift` - Loop selection (range)
- `alt` - Always start at the beginning (and if there are selected parts, start at first selection)

> Of course, these are only the modifiers that I've thought would work well, but if there are better ones, feel free to point them out and suggest them as well.

For the shortcuts behavior:
- `alt`
  - For all playback shortcuts behavior, including `alt` will always start at the beginning of the score or, if there are selected parts, start at the first selection
  - This should be useful when pausing and resuming playback, as there are times when you would want to start at the beginning rather than continuing from the last position
- [`alt` +] `space`
  - Play all unmuted/solo parts regardless of any part measure selection
  - If `alt` is included, start at the beginning of the score or the beginning of the selection if any
- `ctrl` + [`alt` +] `space`
  - Play only parts with selection (if there is no selection, play as `space` behavior)
  - Should continue to play even outside of selection (if any)
- `shift` + [`alt` +] `space`
  - Play and loop selection for all unmuted/solo parts (if there is no selection, play as `space` behavior but loop the entire score or on the loop markers if any)
  - Should not update existing loop markers if loop is enabled. The `shift` modifier should be an override instead and be given higher priority over loop markers
  - If there are existing loop markers and loop is enabled, pausing this playback mode should restore these loop markers
  - Should not continue to play outside of selection (if any) and should start at the beginning of selection if playback is resumed outside of selection
- `ctrl` + `shift` + [`alt` +] `space`
  - Play and loop only parts with selection (if there is no selection, play as `space` behavior but loop the entire score or on the loop markers if any)
  - Same behavior from `shift` + `space` applies (override loop markers, start at beginning when resumed outside of selection if any)

Note that the user should also be able to pause and resume at any point with any or no playback modifiers, and it should continue from the last position unless `alt` is included.

### Problem to be solved

Implementing these playback behavior changes, in my opinion, will give users more control over how they want to start the playback. Also related to https://github.com/musescore/MuseScore/issues/23831, you would be able to select measures and play only those selected measures in a loop just by adding modifiers without additional setup/clicking.

### Checklist

- [X] This request follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting issues
- [X] I have verified that this feature request has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar requests

**Comment by skirby14** [comment]
This sounds fantastic! I've found it surprisingly difficult to adjust to the new behaviour of only playing selected parts in MS4, so something like this would be extremely welcome.

**Comment by MarcSabatella** [comment]
I like the idea of the modifiers.  For discoverability, these could also be selected via a dropdown on the play button.

I know the original concern was that the new select-to-solo feature would not be likely to be discovered quickly if it weren't the default.  Frankly, I'm fine with that, but one way I could see to combine all of this would be to make the default Play behavior be the current one, but have the dropdown on the play toolbar change it to one of the other behaviors.  Kind of like how the note input button works, where N always toggles note input mode, but the dropdown lets you select which specific behavior note input mode implements.

**Comment by MarcSabatella** [comment]
Yes, to be clear, I would like to see *both* the dropdown *and* shortcuts.

Btw, my usual workaround for times when I accidentally select a full measure is to then press Esc.  This clears the selection and allows all instruments to play, but remembers the selected start position.

**Comment by Arnesfield** [comment]
Yes `esc` should also work. However, it feels more like an additional step and it makes select-to-solo an _opt out_ rather than an _opt in_ feature.

The one problem with the modifiers idea is that it's not _backwards compatible_ in the sense that users who use and _expect_ select-to-solo as it is now will feel like it's a bug if it suddenly stops working after an update if the playback modifiers do get implemented and with the behavior changes I've listed out.

I'm also thinking of having an option to disable select-to-solo via settings (enabled by default for backwards compatibility), but maybe there are better ways to go about it.

Thanks!

**Comment by Getsomenow425** [comment]
This is a feature I've missed so much from Musescore 3. I agree that select-to-solo being opt out is frustrating and it disrupts my workflow so much that I've stuck to Musescore 3 this entire time. I really hope more people share this sentiment and can get behind this feature.

**Comment by jmwmusic** [comment]
These shortcuts would be very useful for blind users who can't use the mouse.  Managing play back at the moment requires many tabs, arrows, etc.  There has for example been a suggestion for a shortcut to just repeat the current measure #28042.  Shortcuts to loop range would be helpful, as would ones to solo and mute the current staff and restore all, also for the playback speed slider.
